engine/core/src/main/java/org/datacleaner/descriptors/CompositeDescriptorProvider.java

```diff
@@ -0,0 +1,164 @@
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.descriptors;
+
+import org.apache.commons.collections.collection.CompositeCollection;
+import org.datacleaner.api.*;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * DescriptorProvider that provides a composite view of descriptors from 2 delegate providers.
+ */
+public class CompositeDescriptorProvider implements DescriptorProvider {
+
+    private DescriptorProvider delegate1, delegate2;
+
+    public CompositeDescriptorProvider(DescriptorProvider delegate1, DescriptorProvider delegate2) {
+        this.delegate1 = delegate1;
+        this.delegate2 = delegate2;
+    }
+
+    @Override
+    public Collection<AnalyzerDescriptor<?>> getAnalyzerDescriptors() {
+        return new CompositeCollection(new Collection[] {delegate1.getAnalyzerDescriptors(), delegate2.getAnalyzerDescriptors()});
+    }
+
+    @Override
+    public <A extends Analyzer<?>> AnalyzerDescriptor<A> getAnalyzerDescriptorForClass(Class<A> analyzerClass) {
+        AnalyzerDescriptor<A> result = delegate1.getAnalyzerDescriptorForClass(analyzerClass);
+        if(result != null) { return result; }
+        return delegate2.getAnalyzerDescriptorForClass(analyzerClass);
+    }
+
+    @Override
+    public AnalyzerDescriptor<?> getAnalyzerDescriptorByDisplayName(String name) {
+        AnalyzerDescriptor<?> result = delegate1.getAnalyzerDescriptorByDisplayName(name);
+        if(result != null) { return result; }
+        return delegate2.getAnalyzerDescriptorByDisplayName(name);
+    }
+
+    @Override
+    public Collection<TransformerDescriptor<?>> getTransformerDescriptors() {
+        return new CompositeCollection(new Collection[] {delegate1.getTransformerDescriptors(), delegate2.getTransformerDescriptors()});
+    }
+
+    @Override
+    public <T extends Transformer> TransformerDescriptor<T> getTransformerDescriptorForClass(Class<T> transformerClass) {
+        TransformerDescriptor<T> result = delegate1.getTransformerDescriptorForClass(transformerClass);
+        if(result != null) { return result; }
+        return delegate2.getTransformerDescriptorForClass(transformerClass);
+    }
+
+    @Override
+    public TransformerDescriptor<?> getTransformerDescriptorByDisplayName(String name) {
+        TransformerDescriptor<?> result = delegate1.getTransformerDescriptorByDisplayName(name);
+        if(result != null) { return result; }
+        return delegate2.getTransformerDescriptorByDisplayName(name);
+    }
+
+    @Override
+    public Collection<FilterDescriptor<?, ?>> getFilterDescriptors() {
+        return new CompositeCollection(new Collection[] {delegate1.getFilterDescriptors(), delegate2.getFilterDescriptors()});
```

Comment:
Just wondering: Should we do anything to get rid of duplicates here (and the two other methods for transformers and analyzers)? Maybe we should not worry about it yet, but I am just thinking that since the "get by name" methods will intrinsically only return a single component descriptor, maybe it's weird to provide multiple here?